doc/ot_api_doc.h

```diff
@@ -60,6 +60,7 @@
  * @defgroup api-srp                  SRP
  * @defgroup api-ping-sender          Ping Sender
  * @defgroup api-tcp                  TCP
+ * @defgroup api-tcp-ext              TCP Abstractions
```

Comment:
@jwhui Should we group these together like we do UDP?

Reply:
Yes, good idea. I think we should group these together.

Reply:
@jwhui @Vyrastas No problem, the group is added and tested.